packages/lesswrong/server/callbacks.ts

```diff
@@ -178,6 +181,45 @@ export async function userDeleteContent(user: DbUser, deletingUser: DbUser) {
 
     await commentReportPurgeAsSpam(comment);
   }
+
+  const tags = await Tags.find({userId: user._id}).fetch()
```

Comment:
This seems like it might be the thing you want for the currently-envisaged use-case (i.e. delete everything from a spammer), but it seems like you might want more fine-grained control for other users. E.g. if I request a GDPR removal of my account, but have otherwise been an upstanding forum citizen prior, you might not want to delete my tags, because they might have become part of the general Forum infrastructure and be in use by other users. Ditto wiki revisions. (TBH I'm not 100% clear why tags have a `userId` property as they seem to be more 'part of the commons' than something like a post etc, but presumably there's a good reason.) 

Should there be a flag that toggles between 'delete literally everything' and 'delete the user's personal stuff, but leave alone contributions to the forum commons'?

Reply:
Maybe instead of a flag, you want two methods, `deleteUserContent` (unmodified) and `deleteUserTagsAndRevisions` (for your new code). Might also be nice in terms of readability, as this function is getting pretty long.